styleguide/derek/examples/subpanels.ejs

```diff
@@ -0,0 +1,158 @@
+      <input type="reset" class="btn btn-xs btn-info" id="reset-mini-panels" value="Back to Top"/>
+    </div>
+  </div>
+</div>
+
+<%
+for (var layout in layouts) {
+  for (var pattern_left in patterns) {
+    for (var pattern_right in patterns) {
+      var anchor_id = 'pattern_' + layout + '_' + pattern_left + '_' + pattern_right;
+      var class_left = layouts[layout].class_left;
+      var class_right = layouts[layout].class_right;
+      var section_title = layouts[layout].title + ' (' + patterns[pattern_left].title + '/' + patterns[pattern_right].title + ')';
+      var file_left = '../_partials/solutions/' + pattern_left;
+      var file_right = '../_partials/solutions/' + pattern_right;
+%>
+
+<div id="<%- anchor_id %>" class="container half-padding">
+  <h4><%- section_title %></h4>
+  <div class="row">
+    <div class="<%- class_left %>">
+      <%- partial(file_left, {isSubpanel: true}) %>
+    </div>
+    <div class="<%- class_right %>">
+      <%- partial(file_right, {isSubpanel: true}) %>
+    </div>
+  </div>
+</div>
+<hr/>
+<%
+    }
+  }
+}
+%>
+
+<script type="text/javascript">
+(function ($) {
+  var selectorHeight = 0;
+  var initialOffset = 0;
+  var lastPosition = '';
+  var $patternSelector = $('#pattern-selector');
+  var $body = $('body');
+  var $body_html = $('body, html');
+  var $layout = $('#mp-layout');
+  var $pattern_left = $('#mp-pattern-left');
+  var $pattern_right = $('#mp-pattern-right');
+
+
+  // On scroll, check if we're scrolled past the selector. If so, fix the position to the top.
+  $(window).scroll(function (e) {
+    var scrollTop = $(document).scrollTop();
+    if (scrollTop > initialOffset) {
+      if (lastPosition != 'fixed') {
+        $patternSelector.css({'position': 'fixed', 'top': 0, 'z-index': 100000, 'width': '100%'});
+        $body.css({'padding-top': selectorHeight});
+        lastPosition = 'fixed';
+      }
+    }
+    else {
+      if (lastPosition != 'relative') {
+        $patternSelector.css({'position': 'relative'});
+        $body.css({'padding-top': 0});
+        lastPosition = 'relative';
+      }
+    }
+  });
+
+  // Initialize our selector height and offset and build button behavior.
+  $(document).ready(function () {
+    selectorHeight = parseInt($patternSelector.css('height'), 10);
+    initialOffset = parseInt($patternSelector.offset().top, 10);
+    $(window).trigger('scroll');
+
+    $('#reset-mini-panels').click(function (e) {
+      $body_html.animate({scrollTop: 0}, 250)
+    });
+
+    $('#search-mini-panels').click(function (e) {
+      var layout = $layout.val();
+      var pattern_left = $pattern_left.val();
+      var pattern_right = $pattern_right.val();
+      var id = '#pattern_' + layout + '_' + pattern_left + '_' + pattern_right;
+
+      $body_html.animate({scrollTop: $(id).offset().top - selectorHeight}, 250);
+    });
+  });
+})(jQuery);
+</script>
```

Comment:
Since this is file-specific JS behavior, I figured leaving it here was probably okay.